clientconn.go

```diff
@@ -133,12 +141,31 @@ func (cc *ClientConn) resetTransport(closeTransport bool) error {
 		if closeTransport {
 			t.Close()
 		}
-		newTransport, err := transport.NewClientTransport(cc.dopts.protocol, cc.target, cc.dopts.authOptions)
+		// Adjust timeout for the current try.
+		dopts := cc.dopts
+		if dopts.Timeout > 0 {
+			dopts.Timeout -= time.Since(start)
```

Comment:
dopts is just used to pass the latest timeout value to transport.NewClientTransport when for loop iterates (each iteration needs to have different timeout value for NewClientTransport when time goes forward). The cutoff is start.Add(cc.dopts.Timeout) always.